fixes/general.js

```diff
@@ -210,13 +210,24 @@ function setInternshipAdministrationImprovements(match) {
  * @param {RegExpExecArray} match
  */
 function setPageUserImprovements(match) {
-	// Sort marks listed by project name
+	// Sort marks listed by project name or by completion date
 	const projectItemsContainer = document.querySelector("#marks .overflowable-item");
 	if (projectItemsContainer) {
 		const projectItems = projectItemsContainer.querySelectorAll(".main-project-item, .collapsable");
 		const projectItemsArray = Array.from(projectItems);
-		projectItemsArray.sort((a, b) => {
-			return a.querySelector(".marked-title > a").textContent.localeCompare(b.querySelector(".marked-title > a").textContent);
+		improvedStorage.get("sort-projects-date").then(function(data) {
```

Comment:
This is a good approach, adding this setting here. It needs to be added to the improved_intra_server repository also - including the database model. It will also need screenshots for the settings interface. I can do this, if you focus on the frontend code. You're almost there 😄